keeper-gradle-plugin/src/main/java/com/slack/keeper/InferAndroidTestKeepRules.kt

```diff
@@ -87,13 +98,27 @@ public abstract class InferAndroidTestKeepRules : JavaExec() {
     }
 
     enableAssertions = enableAssertionsProperty.get()
-    standardOutput = outputProguardRules.asFile.get().outputStream().buffered()
-    args = listOf(
-        "--keeprules",
-        classpathJar.get().asFile.absolutePath,
-        appJar.get().asFile.absolutePath,
-        androidTestJar.get().asFile.absolutePath
-    )
+    args = when (traceReferencesEnabled.get()) {
+      false -> listOf(
+          "--keeprules",
+          androidLib.get().asFile.absolutePath,
+          appTargetJar.get().asFile.absolutePath,
+          androidTestSourceJar.get().asFile.absolutePath
+      ).also {
+        // print-uses is using its output to print rules
+        standardOutput = outputProguardRules.asFile.get().outputStream().buffered()
+      }
+      true -> listOf<Pair<String, String?>>(
+          "--keep-rules" to "",
+          "--lib" to androidLib.get().asFile.absolutePath,
+          "--lib" to androidTestLib.orNull?.asFile?.absolutePath,
+          "--target" to appTargetJar.get().asFile.absolutePath,
+          "--source" to androidTestSourceJar.get().asFile.absolutePath,
+          "--output" to outputProguardRules.get().asFile.absolutePath
+      ).map { if (it.second != null) listOf(it.first, it.second) else listOf() }
+          .reduce { acc, any -> acc + any } + traceReferencesArgs.getOrElse(listOf())
+    }
```

Comment:
This isn't very readable, can we comment this a bit or extract a function?

Reply:
Extract as independent methods has drastically increased readability.
Please let me know if it's better now.